src/pool/pool_jemalloc.c

```diff
@@ -0,0 +1,345 @@
+    if (ret != UMF_RESULT_SUCCESS) {
+        return NULL;
+    }
+
+    *zero = true;
+    *commit = true;
+
+    return ptr;
+}
+
+static void arena_extent_destroy(extent_hooks_t *extent_hooks, void *addr,
+                                 size_t size, bool committed,
+                                 unsigned arena_ind) {
+    (void)extent_hooks; // unused
+    (void)committed;    // unused
+
+    jemalloc_memory_pool_t *pool = get_kind_by_arena(arena_ind);
+
+    enum umf_result_t ret = umfMemoryProviderFree(pool->hProvider, addr, size);
+
+    if (ret != UMF_RESULT_SUCCESS) {
+        fprintf(stderr, "umfMemoryProviderFree failed \n");
+    }
+}
+
+static bool arena_extent_dalloc(extent_hooks_t *extent_hooks, void *addr,
+                                size_t size, bool committed,
+                                unsigned arena_ind) {
+    (void)extent_hooks; // unused
+    (void)addr;         // unused
+    (void)size;         // unused
+    (void)committed;    // unused
+    (void)arena_ind;    // unused
+
+    return true;
+}
+
+static bool arena_extent_commit(extent_hooks_t *extent_hooks, void *addr,
+                                size_t size, size_t offset, size_t length,
+                                unsigned arena_ind) {
+    (void)extent_hooks; // unused
+    (void)addr;         // unused
+    (void)size;         // unused
+    (void)offset;       // unused
+    (void)length;       // unused
+    (void)arena_ind;    // unused
+
+    return false;
+}
+
+static bool arena_extent_decommit(extent_hooks_t *extent_hooks, void *addr,
+                                  size_t size, size_t offset, size_t length,
+                                  unsigned arena_ind) {
+    (void)extent_hooks; // unused
+    (void)addr;         // unused
+    (void)size;         // unused
+    (void)offset;       // unused
+    (void)length;       // unused
+    (void)arena_ind;    // unused
+
+    return true;
+}
+
+static bool arena_extent_purge_lazy(extent_hooks_t *extent_hooks, void *addr,
+                                    size_t size, size_t offset, size_t length,
+                                    unsigned arena_ind) {
+    (void)extent_hooks; // unused
+    (void)size;         // unused
+
+    jemalloc_memory_pool_t *pool = get_kind_by_arena(arena_ind);
+    enum umf_result_t ret = umfMemoryProviderPurgeLazy(
+        pool->hProvider, (char *)addr + offset, length);
+
+    return (ret == UMF_RESULT_SUCCESS);
+}
+
+static bool arena_extent_purge_forced(extent_hooks_t *extent_hooks, void *addr,
+                                      size_t size, size_t offset, size_t length,
+                                      unsigned arena_ind) {
+    (void)extent_hooks; // unused
+    (void)size;         // unused
+
+    jemalloc_memory_pool_t *pool = get_kind_by_arena(arena_ind);
+    enum umf_result_t ret = umfMemoryProviderPurgeForce(
+        pool->hProvider, (char *)addr + offset, length);
+
+    return (ret == UMF_RESULT_SUCCESS);
```

Comment:
The same issue with return value as in `arena_extent_purge_lazy` function

Reply:
@vinser52 Done